protocols/migration/migration_reports_4_0_0_to_reports_3_0_0.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+from protocols import reports_4_0_0 as reports_4_0_0
+from protocols import reports_3_0_0 as reports_3_0_0
+from protocols.migration.base_migration import BaseMigration
+from protocols.migration.base_migration import MigrationError
+
+
+class MigrateReports400To300(BaseMigration):
+
+    old_model = reports_4_0_0
+    new_model = reports_3_0_0
+
+    def migrate_clinical_report_rd(self, old_instance):
+        """
+        :type old_instance: reports_4_0_0.ClinicalReportRD
+        :rtype: reports_3_0_0.ClinicalReportRD
+        """
+        new_instance = self.convert_class(self.new_model.ClinicalReportRD, old_instance)
+
+        # interpretationRequestId changed to interpretationRequestID
+        new_instance.interpretationRequestID = old_instance.interpretationRequestId
+
+        # interpretationRequestAnalysisVersion can be null in version 4
+        if hasattr(old_instance, 'interpretationRequestAnalysisVersion') and old_instance.interpretationRequestAnalysisVersion is not None:
+            new_instance.interpretationRequestAnalysisVersion = old_instance.interpretationRequestAnalysisVersion
+        else:
+            new_instance.interpretationRequestAnalysisVersion = old_instance.interpretationRequestVersion
+
+        if hasattr(old_instance, 'candidateVariants') and old_instance.candidateVariants is not None:
+            new_instance.candidateVariants = self.migrate_reported_variants(old_variants=old_instance.candidateVariants)
+        if hasattr(old_instance, 'candidateStructuralVariants') and old_instance.candidateStructuralVariants is not None:
+            new_instance.candidateStructuralVariants = self.migrate_reported_structural_variants(old_structural_variants=old_instance.candidateStructuralVariants)
+
+        return self.validate_object(object_to_validate=new_instance, object_type=self.new_model.ClinicalReportRD)
+
+    def migrate_reported_variants(self, old_variants):
+        return [self.migrate_reported_variant(old_variant) for old_variant in old_variants]
```

Comment:
I think the check for `None` values should be in this function

Reply:
Where ? And why ?

Reply:
Where? In this function before trying to iterate the `old_variants`
Why? If you want to improve code reusability in general and avoid misusage, sanity checks about input is inside functions not on the user side.